packages/block-library/src/template-part/edit/index.js

```diff
@@ -117,12 +139,28 @@ export default function TemplatePartEdit( {
 		[ templatePartId, attributes.area, clientId ]
 	);
 
+	const { templateParts } = useAlternativeTemplateParts(
+		area,
+		templatePartId
+	);
+	const blockPatterns = useAlternativeBlockPatterns( area, clientId );
+	const hasReplacements = !! templateParts.length || !! blockPatterns.length;
```

Comment:
This sort of reverted the changes I made here: https://github.com/WordPress/gutenberg/pull/57856. I'll make a PR to fix it.